src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const emptyArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        emptyArray.push(Object.assign({}, stateCopy));
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        emptyArray.push(Object.assign({}, stateCopy));
+        break;
+
+      case 'clear':
+        for (const stateKey in stateCopy) {
+          delete stateCopy[stateKey];
+        }
+        emptyArray.push(Object.assign({}, stateCopy));
+        break;
+
+      default:
+        throw new Error('Whoops, something bad happened');
+    }
+  }
+
+  return emptyArray;
```

Comment:
or better yet, instead of pushing these values to an array you could just return a map statement:
<img width="647" alt="image" src="https://github.com/mate-academy/js_stateful-clones/assets/87986955/d46a91f1-76e2-4ae0-8b43-bb91a946f2c3">

Reply:
Okay! Thanks for your suggestions :)